python/tests/test_transforms.py

```diff
@@ -874,7 +874,7 @@ def test_projection_truncate_long_literal_eq(bound_reference_long: BoundReferenc
 def test_projection_truncate_long_literal_gt(bound_reference_long: BoundReference) -> None:
     assert TruncateTransform(2).project(
         "name", BoundGreaterThan(term=bound_reference_long, literal=DecimalLiteral(Decimal(19.25)))
-    ) == GreaterThanOrEqual(term="name", literal=Decimal("20.24"))
+    ) == GreaterThanOrEqual(term="name", literal=Decimal("19.26"))
```

Comment:
I think this test is incorrect because the reference is for a decimal column. The adjustment to the literal is probably correct, but we should never have a case where a bound expression contains a term that doesn't match the literal.

Reply:
I'm sorry, I don't understand the issue. It looks equivalent to the Java implementation: https://github.com/apache/iceberg/blob/master/api/src/main/java/org/apache/iceberg/transforms/ProjectionUtil.java#L141-L145